applications/RANSModellingApplication/custom_processes/solving_strategies/scalar_co_solving_process.h

```diff
@@ -0,0 +1,418 @@
+    }
+
+    /// Execute method is used to execute the ScalarCoSolvingProcess algorithms.
+    void Execute() override
+    {
+        if (mrModelPart.GetProcessInfo()[IS_CO_SOLVING_PROCESS_ACTIVE])
+            SolveSolutionStep();
+    }
+
+    virtual int Check() override
+    {
+        KRATOS_CHECK_IS_FALSE(!mrModelPart.HasNodalSolutionStepVariable(this->mrConvergenceVariable));
+
+        for (auto strategy : mrSolvingStrategiesList)
+            strategy->Check();
+
+        for (Process::Pointer auxiliary_process : mAuxiliaryProcessList)
+            auxiliary_process->Check();
+
+        KRATOS_ERROR_IF(mrSolvingStrategiesList.size() == 0)
+            << "No strategies are found for ScalarCoSolvingProcess.";
+
+        return 0;
+    }
+
+    virtual void ExecuteInitialize() override
+    {
+        for (Process::Pointer auxiliary_process : mAuxiliaryProcessList)
+            auxiliary_process->ExecuteInitialize();
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const override
+    {
+        return "ScalarCoSolvingProcess";
+    }
+
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "ScalarCoSolvingProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrModelPart;
+    int mEchoLevel;
+    bool mIsCoSolvingProcessActive;
+    ///@}
+    ///@name Operations
+    ///@{
+
+    virtual void UpdateBeforeSolveSolutionStep()
+    {
+        KRATOS_ERROR << "Calling the base class "
+                        "ScalarCoSolvingProcess::UpdateBeforeSolveSolutionStep."
+                        " Please override it in derrived class.";
+    }
+
+    virtual void UpdateAfterSolveSolutionStep()
+    {
+        KRATOS_ERROR << "Calling the base class "
+                        "ScalarCoSolvingProcess::UpdateAfterSolveSolutionStep. "
+                        "Please override it in derrived class.";
+    }
+
+    virtual void UpdateConvergenceVariable()
+    {
+        KRATOS_ERROR << "Calling the base class "
+                        "ScalarCoSolvingProcess::UpdateConvergenceVariable. "
+                        "Please override it in derrived class.";
+    }
+
+    void ExecuteAuxiliaryProcesses()
+    {
+        for (Process::Pointer auxiliary_process : mAuxiliaryProcessList)
+            auxiliary_process->Execute();
+    }
+
+    void ExecuteAuxiliaryProcessesInitializeSolutionStep()
+    {
+        for (Process::Pointer auxiliary_process : mAuxiliaryProcessList)
+            auxiliary_process->ExecuteInitializeSolutionStep();
+    }
+
+    void SolveSolutionStep()
```

Comment:
can you rename this please?
This is confusing, usually a process does not have this

Reply:
done